packages/block-library/src/search/style.scss

```diff
@@ -55,11 +55,12 @@ $button-spacing-y: math.div($grid-unit-15, 2); // 6px
 		margin-left: 0;
 		// Prevent unintended text wrapping.
 		flex-shrink: 0;
-		// Ensure minimum input field width in small viewports.
+		max-width: 100%;
+	}
+
+	// Ensure minimum input field width in small viewports.
+	.wp-block-search__button[aria-expanded="true"] {
```

Comment:
When in a non button-only state, it looks like there is no `aria-expanded` attribute on the block, so the `max-width: calc(100% - 100px)` rule no longer applies 🤔

<img width="1260" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/897ef499-8177-4f93-a179-470e4601937d">

Reply:
Thanks @andrewserong !

> When in a non button-only state, it looks like there is no aria-expanded attribute on the block, so the max-width: calc(100% - 100px) rule no longer applies

The rules are nested under `.wp-block-search.wp-block-search__button-only`, so this is intentional. 

Are there any issues in non button-only states? Sorry if I missed them!

![2023-09-27 11 26 00](https://github.com/WordPress/gutenberg/assets/6458278/b2d40060-d80e-4ae2-b969-bb70172b114f)

Reply:
> I probably missed some context, there, my apologies! 🤦 This isn't a blocker for this PR, then 🙂

I trust your judgement though. If you think I'm just introducing another bug to fix, I'd rather try to catch it by the tail here! 

I'm mainly trying to build on @t-hamano's work in https://github.com/WordPress/gutenberg/pull/53373, which focussed on the button-only view.

Reply:
> I trust your judgement though. If you think I'm just introducing another bug to fix, I'd rather try to catch it by the tail here!

No, no! I was testing beyond the scope of both this PR and #53373 and reflecting on the minimum size of the input field in the non-button-only state. Now that I take a step back from attempting to break a PR during review 😄 it is incredibly unlikely that someone will use a very, very long text description on the search button in its non-button-only state. I.e. the following:

<img width="324" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/926e68c4-6b92-4337-9807-e0698170d99a">

So while in that situation the input field no longer has a minimum width of `100px`, in practice, I doubt anyone will ever encounter the problem.

I'd proceed to merge this PR and not worry about it for now 🙂

I appreciate the double-checking though! 🙇